subsys/net/ip/net_context.c

```diff
@@ -816,6 +816,24 @@ NET_CONN_CB(tcp_established)
 				     sys_get_be32(NET_TCP_HDR(pkt)->ack));
 	}
 
+//printk("%u %u\n", sys_get_be32(NET_TCP_HDR(pkt)->seq), context->tcp->send_ack);
+
+	/* Peer sent us packet we've already seen. Apparently, our ack was lost. */
+	if (sys_get_be32(NET_TCP_HDR(pkt)->seq) < context->tcp->send_ack) {
+printk("Received already seen pkt - ACKing\n");
+
+	set_appdata_values(pkt, IPPROTO_TCP);
+	uint16_t data_len = net_pkt_appdatalen(pkt);
+
+		uint32_t save = context->tcp->send_ack;
+		context->tcp->send_ack = sys_get_be32(NET_TCP_HDR(pkt)->seq) + data_len;
+		/* Force sending ACK. */
+		context->tcp->sent_ack = context->tcp->send_ack - 1;
+		send_ack(context, &conn->remote_addr);
```

Comment:
We could add a force option into send_ack() call so we could avoid tweaking the sent_ack variable.

Also noticed that the send_ack() is inline, we should remove that although compiler is probably smart enough to do it itself.